community/contributing/pr_review_guidelines.rst

```diff
@@ -0,0 +1,335 @@
+the areas directly affected by PR need to be evaluated more carefully.
+Before going after the code it always makes sense to check if the
+desired outcome is achieved in practice.
+
+.. note::
+	Some PRs can also aim to improve code that could theoretically cause a
+	problem. Use your better judgement and experience there to assess the
+	case, if it cannot be reproduced and tested in practice.
+
+The precise approach would highly depend on the affected area of the
+engine.
+
+4. Do a code review
+^^^^^^^^^^^^^^^^^^^
+
+While code reviews can be quite boring and tedious, we cannot avoid
+them. Thankfully, some PRs are short and sweet, and can probably be
+approved just from the code review alone, skipping steps 1 through 3.
+(Not recommended).
+
+Remember, that a good code review is not only beneficial for the
+codebase itself, but also for the reviewee and the reviewer too. You
+make better contributors out of PR authors when you give them a good
+thoughtful review, or even when you just offer small tips. And you
+can learn something new yourself.
+
+While a proper code review requires you to understand the nuance of the
+affected engine area and is better left for your judgement, there is a
+checklist of universal things to look for:
+
+* **The PR follows** :ref:`doc_code_style_guidelines`.
+
+  While ``clang-format`` and various CI checks can catch a lot of
+  inconsistencies, they are far from perfect in that regard and are
+  unable to detect some issues. For example, check that:
+
+  * The style of header includes is respected;
+  * Identifiers use ``snake_case`` and follow our naming conventions;
+  * Method parameters start with ``p_*`` or ``r_*``.
+
+* **Code only touches the areas announced in the PR (and the commit
+  message).**
+
+  It is always tempting to start fixing random things in code, as you
+  see them. However, this can quickly become a hell to dig through in
+  the commit history. Small touch-ups next to the related area are
+  alright, but often bugs that you can find along the way are better
+  left for their own PRs.
+
+* **Code properly uses Godot's own APIs and patterns.**
+
+  This is true for any reasonably sized project — consistency is very
+  important, and a solution that already exists in the codebase
+  is preferable to an ad-hoc solution.
+
+* **Are core areas affected by the change?**
+
+  Sometimes a PR that is supposed to solve a local problem can have a
+  far-reaching effect, way outside of its scope. If you believe that
+  is the case, make sure to get senior maintainers involved in the
+  process, namely `reduz <https://github.com/reduz>`_ or
+  `vnen <https://github.com/vnen>`_. Their input and approval would
+  be required for changes to the engine's core systems and overall
+  architecture, APIs that touch on all Objects or Nodes, changes
+  to GDScript and GDNative. If the pull request modifies the build
+  system, you can contact `Akien <https://github.com/akien-mga>`_.
```

Comment:
Is hard-coding people into documentation a good idea? Maybe it could link to their teams instead?

Reply:
They were mentioned by name in the notes. Well, reduz and vnen were. I've added akien himself as it seems appropriate.

The recommendation here is specifically for issues that go beyond the teams and need an approval from the leads personally.